nilearn/datasets/tests/test_func.py

```diff
@@ -362,3 +363,48 @@ def test_fetch_mixed_gambles():
         assert_equal(mgambles["zmaps"][0], os.path.join(datasetdir, "zmaps",
                                                         "sub001_zmaps.nii.gz"))
         assert_equal(len(mgambles["zmaps"]), n_subjects)
+
+
+def test_wrong_inputs_of_dimensionality_timeseriesmethods_matrices():
+
+    message = "Invalid %s name"
+
+    dimensionality = ['a10', 'd15', 'd30']
+    assert_raises_regex(ValueError,
+                        message % 'dimensionality',
+                        func.fetch_megatrawls_netmats,
+                        dimensionality=dimensionality)
+
+    timeseries = ['ts4', 'st2', 'eigen_regresion', 'multiple_sptial_regresion']
+    assert_raises_regex(ValueError,
+                        message % 'timeseries',
+                        func.fetch_megatrawls_netmats,
+                        timeseries=timeseries)
+
+    matrices = ['partal_corelation', 'sorelation']
+    assert_raises_regex(ValueError,
+                        message % 'matrices',
+                        func.fetch_megatrawls_netmats,
+                        matrices=matrices)
+
+    assert_raises(TypeError, func.fetch_megatrawls_netmats,
+                  matrices='partial_correlation')
+
+
+@with_setup(tst.setup_tmpdata, tst.teardown_tmpdata)
+def test_fetch_megatrawls_netmats():
+    for dim in [25, 100, 200, 300]:
+        files_dir = os.path.join(tst.tmpdir, 'Megatrawls', '3T_Q1-Q6related468_MSMsulc_d%d_ts3' % dim)
+        os.makedirs(files_dir)
+        dummy = open(os.path.join(files_dir, 'Znet2.txt'), 'w')
```

Comment:
You can use with `open(...) as` to simplify slightly the code (you don't need the .close line)